components/brave_wallet/common/hex_utils.h

```diff
@@ -69,6 +71,48 @@ bool PrefixedHexStringToBytes(std::string_view input,
 std::optional<std::vector<uint8_t>> PrefixedHexStringToBytes(
     std::string_view input);
 
+// Parse a hex string into a fixed-sized buffer, failing if the hex string's
+// size doesn't perfectly match the provided fixed size. This function works
+// in the presence of a leading `0x`, but will work equally fine if the string
+// does not contain it. If required, this function will append a leading 0 if
+// the supplied hex string contains an odd-number of characters (i.e. 0x123 will
+// be treated as 0x0123).
+template <size_t N>
+std::optional<std::array<uint8_t, N>> PrefixedHexStringToFixed(
+    std::string_view input) {
```

Comment:
I'm a bit torn about exposing the whole of this function in the header. Technically, the only thing we need in the header is the array declaration, and the rest we can consign to some `internal::WritePrefixedHexStringToFixed` function that takes a mutable span for the buffer the same way `base::HexStringToSpan` does.